assets/src/less/blocks/wp-block-button/layout.less

```diff
@@ -1,29 +1,94 @@
 /**
  * .wp-block-button
  */
-
 .wp-block-button {
 
+	margin-inline-end: @space-1;
+	margin-block-end: @space-1;
+
+	&:last-child {
+		margin-block-end: 0;
+	}
+
 	.wp-block-button__link {
 		.button();
 	}
 
-	&.alignleft {
+	&.has-custom-width {
+
+		max-inline-size: none;
+
 		.wp-block-button__link {
-			float: inline-start;
-			margin-inline-end: @space-2;
+			inline-size: 100%;
+		}
+
+		&.wp-block-button__width-25 {
+			inline-size: calc(25% - @space-1);
+		}
+
+		&.wp-block-button__width-50 {
+			inline-size: calc(50% - @space-1);
+		}
+
+		&.wp-block-button__width-75 {
+			inline-size: calc(75% - @space-1)
+		}
+
+		&.wp-block-button__width-100 {
+			inline-size: 100%;
+			margin-inline-end: 0;
+		}
+	} 
+} 
+
+/**
+ * .wp-block-buttons
+ */
+.wp-block-buttons {
+
+	display: flex;
+	flex-direction: row;
+	flex-wrap: wrap;
+
+	&.is-content-justification-left {
+		justify-content: flex-start;
+
+		&.is-vertical {
+			align-items: flex-start
 		}
 	}
 
-	&.aligncenter {
-		.text-center;
+	&.is-content-justification-center {
+		justify-content: center;
+
+		.wp-block-button {
+			margin-inline-start: auto;
+			margin-inline-end: auto;
+		}
+
+		&.is-vertical {
+			align-items: center;
+		}
 	}
 
-	&.alignright {
-		.wp-block-button__link {
-			float: inline-end;
-			margin-inline-start: @space-2;
+	&.is-content-justification-right {
+		justify-content: flex-end;
+
+		.wp-block-button {
+			margin-inline-start: @space-1;
+			margin-inline-end: 0;
+		}
+
+		&.is-vertical {
+			align-items: flex-end;
 		}
 	}
 
-} 
\ No newline at end of file
+	&.is-vertical {
```

Comment:
Having this at the bottom of `.wp-block-buttons` overrides `margin-inline-end:auto` in `&.is-content-justification-center` so vertical + center doesn't work.

If you move it up to line 52 it works just fine.